### PR TITLE
[ci] Run check_podspecs on arm Mac bots

### PR DESCRIPTION
**Pull request by jmagman**
Fixes https://github.com/flutter/flutter/issues/141493

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by stuartmorgan — APPROVED** [review]
LGTM, although I would prefer adjusting the PR to add a new arm64 copy, and then remove the x64 copy when taking this out of `bringup`, so that we don't have a coverage gap.

File: .ci.yaml [hunk 1]
```diff
@@ -1062,12 +1062,10 @@ targets:
         }
 
   ### iOS+macOS tasks ###
-  # TODO(stuartmorgan): Move this to ARM once google_maps_flutter has ARM
-  # support. `pod lint` makes a synthetic target that doesn't respect the
-  # pod's arch exclusions, so fails to build.
-  - name: Mac_x64 check_podspecs
+  - name: Mac_arm64 check_podspecs
```
Comment (stuartmorgan) [inline .ci.yaml:1065]: Per discussion in the other issue, you could rename this `repo_checks` now to save another 2-PR cycle later.

Comment (jmagman) [reply .ci.yaml:1065]: Good idea.

**Comment by jmagman** [comment]
> I would prefer adjusting the PR to add a new arm64 copy, and then remove the x64 copy when taking this out of `bringup`, so that we don't have a coverage gap.

Done.

File: .ci.yaml [hunk 2]
```diff
@@ -1073,6 +1071,15 @@ targets:
...
       target_file: macos_check_podspecs.yaml
 
+  - name: Mac_arm64 repo_checks
+    recipe: packages/packages
+    timeout: 30
+    bringup: true
+    properties:
+      add_recipes_cq: "true"
+      version_file: flutter_master.version
+      target_file: macos_check_podspecs.yaml
```
Comment (jmagman) [inline .ci.yaml:1081]: I'll rename to `macos_repo_checks_podspecs.yaml` once `Mac_x64 check_podspecs` is removed and it's not shared.

**Pull request merged**